### PR TITLE
Updated documentation with development notes

### PR DESCRIPTION
**Pull request by chrislovecnm**
It took me a bit to figure out the correct make sequence, so I added a simple note about it.
I also added a new make target for simplicity sake.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -233,6 +233,9 @@ To more cleanly handle cases where desired functionality would require complex d
 
 The root module is generated by running `make generate`. Changes to this repository should be made in the [`autogen`](/autogen) directory where appropriate.
 
+Note: The correct sequence to update the repo using autogen funcationality is the run `make generate && make generate_docs`.  This
```
Comment (morgante) [inline README.md]:
```suggestion
Note: The correct sequence to update the repo using autogen functionality is the run `make generate && make generate_docs`.  This
```

File: README.md [hunk 2]
```diff
@@ -233,6 +233,9 @@ To more cleanly handle cases where desired functionality would require complex d
...
+will create the various terraform files, and then generate the terraform documentation using `terraform-docs`.
```
Comment (morgante) [inline README.md]:
```suggestion
will create the various Terraform files, and then generate the Terraform documentation using `terraform-docs`.
```

**Comment by chrislovecnm** [comment]
@morgante thanks, PTAL

**Comment by chrislovecnm** [comment]
@morgante can we roll this back, or I need to change this.  I did not make the change in the autogen directory.  I am still learning this project layout.

**Pull request merged**